apps/dashboard/src/components/Invoices/List.tsx

```diff
@@ -1,24 +1,137 @@
-import { Timeline, TimelineItem } from "@litespace/luna";
-import { IInvoice } from "@litespace/types";
-import React, { useMemo } from "react";
-import { Hash } from "react-feather";
-import Invoice from "./Invoice";
+import { IInvoice, IWithdrawMethod, Paginated } from "@litespace/types";
+import React from "react";
+import { Table } from "@/components/common/Table";
+import { ActionsMenu, formatCurrency, useFormatMessage } from "@litespace/luna";
+import { createColumnHelper } from "@tanstack/react-table";
+import { useMemo } from "react";
+import DateField from "@/components/common/DateField";
+import { InfiniteData, UseInfiniteQueryResult } from "@tanstack/react-query";
 
-const List: React.FC<{ invoices: IInvoice.Self[]; onUpdate?: () => void }> = ({
-  invoices,
-  onUpdate,
-}) => {
-  const timeline = useMemo((): TimelineItem[] => {
-    return invoices.map((invoice) => ({
-      id: invoice.id,
-      children: <Invoice invoice={invoice} onUpdate={onUpdate} />,
-      icon: <Hash />,
-    }));
-  }, [invoices, onUpdate]);
+const withdrawMethods = {
+  wallet: "withdraw.methods.wallet",
+  instapay: "withdraw.methods.instapay",
+  bank: "withdraw.methods.bank",
+};
+
+const StatusDescriptions = {
+  pending: "invoices.admin.status.pending",
+  "updated-by-receiver": "invoices.admin.status.updatedByReceiver",
+  "canceled-by-receiver": "invoices.admin.status.canceledByReceiver",
+  "cancellation-approved-by-admin": "invoices.admin.status.canceledByAdmin",
+  fulfilled: "invoices.admin.status.fulfilled",
+  rejected: "invoices.admin.status.rejected",
+};
+
+const List: React.FC<{
+  invoicesList: IInvoice.Self[] | null;
+  query: UseInfiniteQueryResult<
+    InfiniteData<Paginated<IInvoice.Self>, unknown>,
+    Error
+  >;
+}> = ({ invoicesList, query }) => {
+  const intl = useFormatMessage();
+  const columnHelper = createColumnHelper<IInvoice.Self>();
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor("id", {
+        header: intl("global.labels.id"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("userId", {
+        header: intl("dashboard.invoices.userId"),
+        cell: (info) => info.getValue(),
+      }),
+      columnHelper.accessor("method", {
+        header: intl("dashboard.invoices.method"),
+        cell: (info) => {
+          const status = info.getValue() as IWithdrawMethod.Type;
+          const value = withdrawMethods[status] as keyof typeof intl;
```

Comment:
Try to avoid casting as much as I can as It is inherently unsafe. Take a look at my comments above to know how to resolve it in a type-safe manner.